driver/src/main/java/org/neo4j/driver/v1/Record.java

```diff
@@ -19,33 +19,8 @@
 package org.neo4j.driver.v1;
 
 /**
- * A record is a collection of named fields, and is what makes up the individual items in a {@link
- * Result}
+ * A record is an immutable record view
  */
-public interface Record
-{
-    /**
-     * From the current record the result is pointing to, retrieve the value in the specified field.
-     *
-     * @param fieldIndex the field index into the current record
-     * @return the value in the specified field
-     */
-    Value get( int fieldIndex );
-
-    /**
-     * From the current record the result is pointing to, retrieve the value in the specified field.
-     * If no value could be found in the specified filed, null will be returned.
-     *
-     * @param fieldName the field to retrieve the value from
-     * @return the value in the specified field or null if no value could be found in the field.
-     */
-    Value get( String fieldName );
-
-    /**
-     * Get an ordered sequence of the field names in this result.
-     *
-     * @return field names
-     */
-    Iterable<String> fieldNames();
-
+public interface Record extends RecordView {
+    // intentionally empty
```

Comment:
To me that is super strange because it breaks substitution. If the super type is immutable but the sub type is mutable then what contract holds if I have a value of the super type? On the other hand, when starting from (potential) mutability there is a clear contract, and a sub type then can narrow this down to immutability.